src/components/App/App.js

```diff
@@ -98,55 +134,90 @@ const App = () => {
   /******************************************************************************************** */
   /** ***************************** Handles `Register` & `Login` Logic *************************** */
 
-  const onRegister = ({ email, password, username }) => {
+  const onRegister = ({ email, password, name }) => {
     auth
-      .register(email, password, username)
+      .register(email, password, name)
       .then((res) => {
-        setIsSuccessOpen(true);
+        if (res.status === 201) {
+          setIsRegisterOpen(false);
+          setIsSuccessOpen(true);
+        }
+
+        if (res.status === 409 || 500 || 400) {
+          setSubmitError(res.message);
+        }
       })
       .catch((err) => console.log(err));
   };
 
```

Comment:
* in `SearchForm` you need to add `value` to the input.  If you have `onChange` or ` onInput` in an input then you need to add attribute `value` to this input.